aiida_quantumespresso/parsers/pp.py

```diff
@@ -140,12 +124,25 @@ def get_key_from_filename(filename):
             matches = re.search(pattern, filename)
             return matches.group(1)
 
-        for filename, data in data_raw:
-            try:
-                key = get_key_from_filename(filename)
-                data_parsed.append((key, parsers[iflag](data)))
-            except Exception:  # pylint: disable=broad-except
-                return self.exit_codes.ERROR_OUTPUT_DATAFILE_PARSE.format(filename=filename)
+        for filename in filenames:
+            if filename.endswith(filename_suffix):
+                try:
+                    with file_opener(filename) as handle:
+                        data_raw = handle.read()
+                except OSError:
+                    return self.exit_codes.ERROR_OUTPUT_DATAFILE_READ.format(filename=filename)
+
+                try:
+                    key = get_key_from_filename(filename)
+                    data_parsed.append((key, parsers[iflag](data_raw)))
```

Comment:
Probably adding a `del data_raw` here could reduce the memory usage further. I think because `handle.read()` is executed before the assignment on `data_raw` above, you can still end up with a case where two objects are in memory simultaneously.